packages/camera/camera/ios/Classes/FLTSavePhotoDelegate.m

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTSavePhotoDelegate.h"
+
+@implementation FLTSavePhotoDelegate
+
+- (instancetype)initWithPath:(NSString *)path
+                      result:(FLTThreadSafeFlutterResult *)result
+                     ioQueue:(dispatch_queue_t)ioQueue {
+  self = [super init];
+  NSAssert(self, @"super init cannot be nil");
+  _path = path;
+  _selfReference = self;
```

Comment:
I know you didn't write this but this is a strange way to keep `self` alive, and it's shared between all the completion blocks, so it could be set to `nil` in one but not before another is complete.  You'd think it would be retained by `FLTCam` or something...
Anyway this is no worse than it was.

Reply:
This is discussed in the "Remove selfReference" part of [the proposal](https://docs.google.com/document/d/1Znjp-BduXIb1Ix3JOE1YKyJi1Z0d6rB-2q6Lt7j8xSo/edit#heading=h.p926krbv9zgw): 

> selfReference is used to keep FLTSavePhotoDelegate around. This is a nice trick, but should only be used as a workaround. In our case this delegate object has a clear owner - the delegating object (FLTCam), which controls the lifecycle of this delegate.

Reply:
Ha well at least we were in agreement.